spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/Session.scala

```diff
@@ -254,27 +255,59 @@ class BigDLSessionImpl[T: ClassTag](graph: Seq[NodeDef], context: Context[T],
     result
   }
 
+  private def allHDFSFiles(fileNames: Seq[String]) = {
+    val isHdfs = fileNames.map(_.startsWith("hdfs:"))
+    if (isHdfs.reduceLeft(_ && _)) {
+      true
+    } else if (isHdfs.map(!_).reduceLeft(_ && _)) {
+      false
+    } else {
+      throw new IllegalArgumentException("filenames contain both local and hdfs path")
+    }
+  }
+
   private def readTFRecord(filesTable: Seq[Table], sc: SparkContext): RDD[Table] = {
-    val result = filesTable.map { t =>
-        require(t.length() == 1 && t(1).isInstanceOf[Tensor[ByteString]],
-          "Reader can only read one file at a time")
-        val fileTensor = t[Tensor[ByteString]](1)
-        require(fileTensor.isScalar)
-        val file = fileTensor.value()
-        file
-    }.flatMap { file =>
-      val iter = new TFRecordIterator(new java.io.File(file.toStringUtf8))
-      iter
-    }.map { record =>
-      val table = T()
-      val key = Tensor[ByteString](Array(ByteString.copyFromUtf8("somekey")), Array[Int]())
-      val value = Tensor[ByteString](Array(ByteString.copyFrom(record)), Array[Int]())
-      table.insert(key)
-      table.insert(value)
-      table
+    if (filesTable.isEmpty) {
+      return sc.parallelize(Seq.empty[Table], numSlices = Engine.coreNumber())
+    }
+    val fileNames = filesTable.map { t =>
+      require(t.length() == 1 && t(1).isInstanceOf[Tensor[ByteString]],
```

Comment:
even if t(1) is Tensor[Float, ]t(1).isInstanceOf[Tensor[ByteString]] will also return true. Try to use the getType to check the numeric type.

Reply:
Two error cause, but in the message you only give one.